crates/relayer/src/chain/ckb4ibc.rs

```diff
@@ -1273,9 +1275,62 @@ impl ChainEndpoint for Ckb4IbcChain {
         Ok((sequence, None))
     }
 
-    fn query_txs(&self, _request: QueryTxRequest) -> Result<Vec<IbcEventWithHeight>, Error> {
-        warn!("ckb4ibc query_txs() not support");
-        Ok(vec![])
+    fn query_txs(&self, request: QueryTxRequest) -> Result<Vec<IbcEventWithHeight>, Error> {
+        let prefix = self.query_commitment_prefix()?;
+        let events = match request {
+            QueryTxRequest::Client(QueryClientEventRequest {
+                query_height: _,
+                event_id: _,
+                client_id: _,
+                consensus_height,
+            }) => {
+                let block = self.rt.block_on(
+                    self.rpc_client
+                        .get_block_by_number(consensus_height.revision_height().into()),
+                )?;
+                let block_number: u64 = block.header.inner.number.into();
+                block
+                    .transactions
+                    .into_iter()
+                    .map(|tx| {
+                        let event = transaction_to_event(&tx, &prefix)?;
```

Comment:
If the CKB block contains `non-IBC-event` transactions, then it may fail here.

Reply:
done, review again